### PR TITLE
Dev aardvark

### PR DESCRIPTION
**Pull request by capolitsch**
aardvark package now compatible with evalcast when using usa-facts signals. still problems with JHU signals?

### CODE REVIEW

**Comment by dajmcdon** [comment]
@capolitsch The R checks are failing. Seems to be an issue with one of the examples. Can you fix it up (good practice is to rerun the R CMD check locally by, for example, clicking the button next "Build" in Rstudio. 

Is this still not working with JHU? October 1 seems to be an unfortunately troublesome date in the API for some reason.

**Comment by capnrefsmmat** [comment]
If there's anything wrong with the JHU data currently in the API (not just that JHU reported something weird on that day, but that our version is wrong), please file an indicators issue -- I don't think engineering is currently tracking any remaining problems after the outage a few weeks ago.

**Comment by dajmcdon** [comment]
I'm not aware of any issues at the moment. There was an issue with October 1, but currently seems fine for cases/deaths, usa-facts/jhu-csse. I'm not sure what Collin's comment above is referring to.

**Comment by capolitsch** [comment]
Thanks for the ping. I'm investigating and will keep you posted.

**Comment by dajmcdon** [comment]
@capolitsch What's the status of this PR?

**Comment by capolitsch** [comment]
@dajmcdon @jsharpna All checks pass and the forecaster is running as expected. Ready for view.

**Comment by dajmcdon** [comment]
@capolitsch How challenging will it be to use the newer `evalcast`?

**Comment by capolitsch** [comment]
@dajmcdon This runs with `evalcast-killcards`. Are there other changes coming to `evalcast`?

**Review by dajmcdon — CHANGES_REQUESTED** [review]
A few minor things, but otherwise, I approve. @jsharpna, this PR merges in zookeeper from the `fcast-utils` branch. How do you feel about that? I don't particularly have a problem.

File: forecasters/aardvark/DESCRIPTION [hunk 1]
```diff
@@ -15,12 +15,12 @@ Description: This package produces the Carnegie Mellon Delphi Lab COVID-19
   state death forecaster.
 License: MIT + file LICENSE
 Encoding: UTF-8
-Suggests: testthat
-Imports: evalcast, lubridate, tidyr, dplyr, utils, purrr, stats,
-        Matrix, glmnet, zoo, magrittr
-Remotes: cmu-delphi/covidcast/R-packages/evalcast@evalcast
+Imports: evalcast, covidcast, lubridate, tidyr, dplyr, utils, purrr, stats,
+         Matrix, glmnet, zoo, magrittr
+Remotes: cmu-delphi/covidcast/R-packages/evalcast@evalcast,
```
[the remark this answers is not in the document]
Comment (capolitsch) [reply forecasters/aardvark/DESCRIPTION:20]: Ahh yes, this is true. This is an oversight, but the only consequence is that it uses the main `evalcast` branch for the calls to `get_target_period`. Externally I am using `evalcast-killcards` for `get_predictions` and `plot_trajectory`.

File: forecasters/aardvark/R/bootstrapping.R [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    # (1) Get conditional mean predictions, on weekly scale.
+    target_dates <- get_target_period(forecast_date, incidence_period, ahead) %$%
+      seq(start,end,by = "days")
+    point_preds <- df_point_preds %>% filter(time_value %in% target_dates) %>%
+      select(location, time_value, preds)
+    stopifnot(nrow(point_preds) == length(unique(point_preds$location)) * length(target_dates))
+    point_preds <- point_preds %>% group_by(location) %>%
+      summarize(preds = sum(preds), .groups = "drop") %>%
+      mutate(time_value = NA)
```
[the remark this answers is not in the document]
Comment (capolitsch) [reply forecasters/aardvark/R/bootstrapping.R]: Fixed

File: forecasters/aardvark/R/modeling.R [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+  names(df3)[which(substr(names(df3),1,5) == "value")] <- "value"
+  df <- bind_rows(df1, df2, df3)
+  match.string.2 <- with(df, paste0(variable_name, geo_value, time_value))
+  df$issue <- df.tmp$issue[match(match.string.2, match.string.1)]
+  df <- df %>% select(location, geo_value, variable_name, value, time_value, issue)
+  df$value <- as.double(df$value)
+  return(df)
+}
+
+#' @importFrom Matrix Matrix
```
[the remark this answers is not in the document]
Comment (capolitsch) [reply forecasters/aardvark/R/modeling.R]: Fixed

File: forecasters/aardvark/R/time_alignment.R [hunk 4]
```diff
@@ -0,0 +1,52 @@
+#-----------------------------------------------------------------------------#
+# This script contains aligners: functions used to compute aligned time.
+# Each aligner should take as input:
+#   df_use: a data frame, of the same form as the upstream df
+#   forecast_date: a canonical parameter used by the evaluator.
+# and output a data frame
+#   df_align: a data frame with columns location, time_value, and align_date.
+# 
+# Each aligner should satisfy the "aligner guarantee".
```
[the remark this answers is not in the document]
Comment (capolitsch) [reply forecasters/aardvark/R/time_alignment.R:9]: Probably still pretty important, but we likely need to update the way we do it. Planning to experiment with this during improvements.

File: forecasters/aardvark/R/modeling.R [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+      top_n(1, wt = issue) %>% # NA only chosen if that's all there is
+      ungroup %>%
+      mutate(issue = na_if(issue, min_issue - 1)) # go back to NA
+  }else{
+    df.tmp <- df
+  }
+  match.string.1 <- with(df.tmp, paste0(data_source, "-", signal, geo_value, time_value))
+  df <- df %>% mutate(variable_name = paste(data_source, signal, sep = "-"),
+                      location = formatC(covidcast::state_census$STATE[match(toupper(df$geo_value),
+                                                                             covidcast::state_census$ABBR)],
```
[the remark this answers is not in the document]
Comment (capolitsch) [reply forecasters/aardvark/R/modeling.R]: Nice. I updated it.

File: forecasters/aardvark/DESCRIPTION (hunk 1 above)
[the remark this answers is not in the document]
Comment (capolitsch) [reply forecasters/aardvark/DESCRIPTION]: Fixed.

**Review by capolitsch — COMMENTED** [review]
All changes addressed

**Review by dajmcdon — APPROVED** [review]

**Pull request merged**